File: src/main/java/com/ulbs/careerstartup/dto/LanguageDTO.java
package com.ulbs.careerstartup.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.ulbs.careerstartup.constant.LanguageLevel;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;

import jakarta.validation.constraints.NotNull;
import java.util.UUID;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class LanguageDTO {

    private UUID id;

    @NotNull(message = "Name is required")
    @NotEmpty(message = "Name is required")
    private String name;

    @NotNull(message = "Listening level is required")
    private LanguageLevel listening;

    @NotNull(message = "Reading level is required")
    private LanguageLevel reading;

    @NotNull(message = "Speaking level is required")
    private LanguageLevel speaking;

    @NotNull(message = "Conversation level is required")
    private LanguageLevel conversation;

    @NotNull(message = "Writing level is required")
    private LanguageLevel writing;

    @JsonProperty("file")
    private FileDTO fileDTO;

    @NotNull(message = "User is required")
    @JsonProperty("user")
    private UserDTO userDTO;
}

File: src/app/services/models/language-dto.ts
/* tslint:disable */
/* eslint-disable */
import { FileDto } from '../models/file-dto';
import { UserDto } from '../models/user-dto';

/**
 * Id of the language that will be deleted
 */
export interface LanguageDto {
  conversation: 'A1' | 'A2' | 'B1' | 'B2' | 'C1' | 'C2' | 'TONGUE';
  file?: FileDto;
  id?: string;
  listening: 'A1' | 'A2' | 'B1' | 'B2' | 'C1' | 'C2' | 'TONGUE';
  name: string;
  reading: 'A1' | 'A2' | 'B1' | 'B2' | 'C1' | 'C2' | 'TONGUE';
  speaking: 'A1' | 'A2' | 'B1' | 'B2' | 'C1' | 'C2' | 'TONGUE';
  user: UserDto;
  writing: 'A1' | 'A2' | 'B1' | 'B2' | 'C1' | 'C2' | 'TONGUE';
}
